sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -325,7 +325,9 @@ case class Invoke(
 
   @transient lazy val method = targetObject.dataType match {
     case ObjectType(cls) =>
-      val m = cls.getMethods.find(_.getName == encodedFunctionName)
+      val m = cls.getMethods.find { m =>
+        m.getName == encodedFunctionName && m.getParameterCount == arguments.length
```

Comment:
should we check parameter types too? also curious why we don't use `cls.getMethod` for the purpose.

Reply:
+1 to use `getMethod`

Reply:
Actually `Invoke` uses `getMethod` for target object with data types other than `ObjectType`. I'm not sure why we need to do a separate `getMethod` during evaluation for the cases.

Reply:
And note `method` uses `encodedFunctionName` to find method, but in `eval` the original `functionName` is used.

Reply:
Switched to use `getMethod` and see what CI tells.

Reply:
It looks like a very ancient bug from Spark 1.x.